spec/plans/sync.fmf

```diff
@@ -0,0 +1,56 @@
+summary: Define synchronization method for each TMT step
+
+description: |
+    The main idea of :ref:`/spec/plans/provision/multihost`
+    tests is to run certain tasks in parallel (e.g., a RestAPI
+    requests from a client to a server). Therefore, these kinds
+    of tests may require test synchronization.
+    The synchronization method may be specified for each step
+    by providing one of defined options under ``step`` keyword.
+    This creates a synchronization barrier before each provided step.
+
+        start
+            Synchronize at the beginning of the ``step`` by adding
+            a synchronization barrier at the beginning of the step.
+            In case of multiple configurations, it also adds a barrier
+            at the beginning of the ``step``, even if the configuration
+            filters out every ``test``.
+        name
+            Synchronize execution of tests with the same name. Standalone
```

Comment:
Does it mean `tmt` needs to implement some heuristics to do the syncing properly? Like having two test sets.
for `server`
```
test1
test-multi-1
test2
test-multi-2
```
and for `client`
```
test-multi-1
test3
test4
test-multi-2
```
`tmt` should somehow realize that the first common match is `test-multi-1` and sync on that? I guess that may become quite complicated when more than 2 servers are involved and there is quite a lot of space for deadlocks. Would you have some specific idea how this would be implemented?